client/web/src/insights/pages/creation/search-insight/components/search-insight-creation-content/validators.ts

```diff
@@ -12,8 +12,17 @@ export const requiredStepValueField = createRequiredValidator('Please specify a
  * Custom validator for chart series. Since series has complex type
  * we can't validate this with standard validators.
  * */
-export const seriesRequired: Validator<DataSeries[]> = series =>
-    series && series.length > 0 ? undefined : 'Series is empty. You must have at least one series for code insight.'
+export const seriesRequired: Validator<EditableDataSeries[]> = series => {
+    if (!series || series.length === 0) {
+        return 'Series is empty. You must have at least one series for code insight.'
```

Comment:
Here's my version (solve ambigous singular/plural in the first sentence, have->add, "for code insight"->"to create a code insight")
```suggestion
        return 'No series defined. You must add at least one series to create a code insight.'
```

Reply:
This wouldn't actually happen in practice in the new logic, right? Because there should always be one series in the array by default, which just may have empty, required fields (in which case we should just only show those errors)? Or in which case would this show?

Reply:
Nice catch, I guess you're right, maybe one corner case which came to my mind about what if a user manually deleted these series from the series array and tries to edit this bad insight. Probably we should add some check there where we serialize insight from user/org settings. 
I create a ticket for corner cases of this validation, so let's fix that in separate PR.

Reply:
Thanks for proofreading